var/spack/repos/builtin/packages/pcre2/package.py

```diff
@@ -43,11 +69,17 @@ def configure_args(self):
 
         return args
 
-    @property
-    def libs(self):
-        if "+multibyte" in self.spec:
-            name = "libpcre2-32"
-        else:
-            name = "libpcre2-8"
 
-        return find_libraries(name, root=self.prefix, recursive=True)
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    def cmake_args(self):
+        args = []
+        args.append(self.define_from_variant("PCRE2_BUILD_PCRE2_16", "multibyte"))
+        args.append(self.define_from_variant("PCRE2_BUILD_PCRE2_32", "multibyte"))
+        args.append(self.define_from_variant("PCRE2_SUPPORT_JIT", "jit"))
+        args.append(self.define_from_variant("BUILD_SHARED_LIBS", "shared"))
+        # PCRE allows building shared and static at the same time
+        # this is bad practice and a problem on some platforms
+        # Enforce mutual exclusivity here
+        args.append(self.define("BUILD_STATIC_LIBS", not self.spec.satisfies("shared")))
```

Comment:
`define_from_variant` can just specify the variant name, but `spec.satisfies` needs a variant value, like `not spec.satisfies(+shared)`.